src/status_im/hardwallet/core.cljs

```diff
@@ -148,11 +150,27 @@
       (unauthorized-operation cofx))))
 
 (fx/defn navigate-to-authentication-method
-  [cofx]
+  [{:keys [db] :as cofx}]
   (if (hardwallet-supported? cofx)
-    (navigation/navigate-to-cofx cofx :hardwallet-authentication-method nil)
+    (fx/merge cofx
+              {:db (assoc-in db [:hardwallet :flow] :create)}
+              (navigation/navigate-to-cofx :hardwallet-authentication-method nil))
     (accounts.create/navigate-to-create-account-screen cofx)))
 
+(fx/defn navigate-to-recover-method
+  [{:keys [db] :as cofx}]
+  (if (hardwallet-supported? cofx)
+    (fx/merge cofx
+              {:db (assoc-in db [:hardwallet :flow] :import)}
```

Comment:
Any reason why this `:flow` parameter couldn't be passed as a navigation parameter, e.g. 

`(navigation/navigate-to-cofx :hardwallet-authentication-method {:flow :import})`

and then get it with `:get-screen-params` subs.

Reply:
@goranjovic `:flow` parameter is used in several different screens. I didn't use `get-screen-params`, but seems like parameters there are related to one particular there if i get it right. In my case, `:flow` is more like global parameter which is used throughout many screens.

Reply:
Fair enough